jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -152,6 +154,44 @@ public boolean isSame(Resource resource)
         return equals(resource);
     }
 
+    /**
+     * Return an Iterator of all Resource's referenced in this Resource.
+     *
+     * <p>
+     *     This is meaningful if you have a Composite Resource, otherwise it will be a single entry Iterator.
+     * </p>
+     *
+     * @return the iterator of Resources.
+     */
+    @Override
+    public Iterator<Resource> iterator()
+    {
+        return new Iterator<>()
```

Comment:
Uhm. `List.of(this).iterator()` seems a lot simpler.

`List.of(this)` could be cached as a field if worried about the extra allocation.

Reply:
Not worried about allocation, as calling  `.iterator()` will allocate anyway.

Reply:
Done.